src/components/AstronautsPage/reducers/astronautsReducer.ts

```diff
@@ -0,0 +1,37 @@
+import { AstronautsActions } from '../actions/actionTypes';
+import type {
+  AstronautsState,
+  AstronautsAction,
+} from '../types/AstronautsAction';
+import type { Astronauts } from '../types/Astronauts';
+
+const initialState: AstronautsState = {
+  astronauts: [],
+  isLoading: false,
+  error: '',
+};
+
+export const astronautsReducer = (
+  state = initialState,
+  { type, payload }: AstronautsAction,
+) => {
+  switch (type) {
+    case AstronautsActions.FETCH_ASTRONAUTS_REQUEST:
+      return {
+        ...state,
+        isLoading: true,
+      };
+    case AstronautsActions.FETCH_ASTRONAUTS_SUCCESS:
+      return {
+        ...state,
+        astronauts: payload as Astronauts[],
+      };
+    case AstronautsActions.FETCH_ASTRONAUTS_FAILURE:
+      return {
+        ...state,
+        error: payload as string,
```

Comment:
why do you need as here?

Reply:
it was Sasha's idea, but i've read your conversation and also fix it